### DISCUSSION TITLE
How to use objectscript.export.addCategory

### DISCUSSION THREAD

**Discussion by mdavidovichHst** [intersystems-community/vscode-objectscript#1279] [Q&A] [answered]
Hello, I'm having trouble with this setting.  My context and reference is https://github.com/intersystems-community/intersystems-iris-dev-template.

- The folder path is structured as /src/CATEGORY/package/ClassName.cls, yes?
- So if I have /src/cls/Roles and add Student.cls to the Roles folder, the file should automatically populate `Class Roles.Student Extends %RegisteredObject`, yes?
- When I set my workspace settings to `objectscript.export.addCategory=false` I get `Class cls.Roles.Student Extends . . .`
- Is this setting embedded somewhere in https://github.com/intersystems-community/intersystems-iris-dev-template that it overrides my workspace setting (I added that repo to my workspace).  

Thanks for your help.  I have read https://intersystems-community.github.io/vscode-objectscript/settings/ but I can't find my answer.

**Comment by isc-bsaviano** [comment, marked as answer]
@mdavidovichHst In your example, you need `addCategory` to be `true`, since your folder structure has the category layer. My personal opinion is that the category folders don't add any value, so if you're starting from scratch I would suggest setting it to `false`. Your classes and routines will be put in the same folder if they're in the same package.

**Comment by mdavidovichHst** [comment]
@isc-bsaviano Thanks, Brett!

This makes sense now, I was thinking about it backwards.

Not entirely starting from scratch here.  Just seeing if I can organize my code into this template to have a sandbox to work with for containerization.  All our routines are in one folder plus I have a ton a CSP files, so I think it still make sense to keep src/cls and src/rtn and src/csp.
